.travis.yml

```diff
@@ -14,32 +14,78 @@ go_import_path: github.com/kubeless/kubeless
+    - secure: CXyDSLR6KJGoPN4ApD4UjIqXxKWxvbSuun1Py3ViI6XuhJOjTBlNJVvCV3cZUD+Dvs98sUnNVvrZgoP/YxCIF6tPzm6VboiO75gxXKJQCDbOuUE9PhBiU2WhHCEXS0AL7emCgKqOw4hNWUdyzhx+lytZkIrFFwuT0qUVqCF/1ccDJXEl3YeMotoMqiV+XQZIvoMNQVY8ENJvYNoBWi+iFF+lSRBUMavB/4soXvZjASMfz189/BQTGfXeOeKu6E+O51zup7Z+W28b9bO1r47jjY8VFdLrnnfDPtN7qEd/An4OkICnijcMHbS4n1NO9mb/wzYvOuo7hDWU5cHDbN/rhaoFmHa/8Qp9N8UMza70ps4mO4jYg+e9ZTGZvFMRU3gcF7XRkrdqBFQOsgTVJrZsfzRFiV5M30PXv4gc7UXVvsHrSj76bRpQL+dkGdqIQZxc9OFAOJ/dRLwGowu1dS5N63444xGx5PuC5IZ81Z4qIDfzpAfM5637NHrkNT9KRFyJktAK8XK1e7c1wD95uFyEDv9UPziJLEPOtTP8/YLSs1H0TgjVAyStv6HcqHDHff5n25rFehNlULpJcktqKv1YEyYGak9nwGsSTk2h73kvKp2cAfLdlW9vN9IxF60G+G+OmB4ycK+KfU0FNg/NVm40+LeogIWgz9Xh3LV+obEeMC0=
 before_install:
   - set -e
 
 install:
   - make bootstrap
-
-script:
-  - make test
-  - make validation
-  - make VERSION=${TRAVIS_TAG:-build-$TRAVIS_BUILD_ID} binary
   - |
-    if [ "$TRAVIS_OS_NAME" = linux ]; then
-      make controller-image CONTROLLER_IMAGE=$CONTROLLER_IMAGE
+    if [[ "$TEST_TARGET" == "GKE" && -n "$GKE_ADMIN" && "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then
+      export GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+      openssl aes-256-cbc -K $encrypted_1dbafa65b3ad_key -iv $encrypted_1dbafa65b3ad_iv -in .github/gcloud.json.enc -out $GOOGLE_APPLICATION_CREDENTIALS -d
+      if [ ! -d $HOME/gcloud/google-cloud-sdk ]; then
+        mkdir -p $HOME/gcloud &&
+        wget -q https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz --directory-prefix=$HOME/gcloud &&
+        cd $HOME/gcloud &&
+        tar xzf google-cloud-sdk.tar.gz &&
+        printf '\ny\n\ny\ny\n' | ./google-cloud-sdk/install.sh &&
+        cd $TRAVIS_BUILD_DIR;
+      fi
+      gcloud -q config set project $GKE_PROJECT
+      if [ -a $GOOGLE_APPLICATION_CREDENTIALS ]; then
+        gcloud -q auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS;
+      fi
+      echo "Starting GKE cluster"
+      ./script/start-gke-env.sh $GKE_CLUSTER $ZONE $GKE_VERSION $GKE_ADMIN > $TRAVIS_BUILD_DIR/gke-start.log 2>&1 &
+      GKE_START_PID=$!
     fi
-  - make all-yaml
+
+script:
   - |
-    if [ "$TRAVIS_OS_NAME" = linux ]; then
+    make VERSION=${TRAVIS_TAG:-build-$TRAVIS_BUILD_ID} binary
+    make controller-image CONTROLLER_IMAGE=$CONTROLLER_IMAGE
+    case $TEST_TARGET in
+    unit)
+      make test
+      make validation
+      ;;
+    minikube)
       export TEST_DEBUG=1
-      docker tag $CONTROLLER_IMAGE ${CONTROLLER_IMAGE_NAME}:latest
       make integration-tests
-    fi
+      ;;
+    GKE)
+      if [[ -n "$GKE_ADMIN" && "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then
```

Comment:
Should we constrain it to the master branch only, like the `after_success` step? With this config, if someone creates a branch on kubeless/kubeless and PRs that, I believe the test will run.

Reply:
That's exactly the goal of this check. Note that if we don't do that we wouldn't be able to test changes in the GKE integration before merging it into master (and that's how I tested this PR).

Reply:
Ah okay, makes sense!